starknet_py/contract.py

```diff
@@ -250,16 +289,28 @@ async def call(
         The result is translated from Cairo data to python values.
         Equivalent of ``.prepare(*args, **kwargs).call()``.
         """
-        return await self.prepare(*args, **kwargs).call(
+        return await self.prepare(max_fee=0, version=0, *args, **kwargs).call(
             block_hash=block_hash, block_number=block_number
         )
 
-    async def invoke(self, *args, **kwargs) -> InvokeResult:
+    async def invoke(
+        self, *args, max_fee: int = 0, auto_estimate: bool = False, **kwargs
+    ) -> InvokeResult:
         """
         Invoke contract's function. ``*args`` and ``**kwargs`` are translated into Cairo calldata.
         Equivalent of ``.prepare(*args, **kwargs).invoke()``.
+
+        :param max_fee: Max amount of Wei to be paid when executing transaction
+        :param auto_estimate: Use automatic fee estimation, not recommend as it may lead to high costs
         """
-        return await self.prepare(*args, **kwargs).invoke()
+        prepared_call = self.prepare(*args, **kwargs)
+
+        if auto_estimate and max_fee is None:
+            estimate_fee = await prepared_call.estimate_fee()
+            max_fee = int(estimate_fee * 1.1)
+
+        prepared_call.max_fee = max_fee
+        return await prepared_call.invoke()
```

Comment:
The same logic is added to prepared call, can't we just call prepared_call.invoke(max_fee) if we don't provide max_fee to it in the first place?

Reply:
I've removed duplicated logic and changed it to `prepared_call.invoke(max_fee, estimate_fee)`